### ISSUE TITLE
A request to add new functions to pygame.transform module

### ISSUE THREAD

**Issue by ankith26** [pygame/pygame#2128]
TLDR; I want to just contribute a few functions (to pygame.transform) that I think will be quite useful to a lot of people. Of course, this is just a small request and if my request is turned down I will be totally ok with it.
Thanks a lot to everyone involved in pygame dev, to give to the world such a wonderful tool. I just want to play my small part and contribute to this community project. 

Now let me begin. 

I was playing around with pygame.tranform, I wanted to scale down some images and stuff. I used `pygame.transform.scale()` and `pygame.transform.smoothscale()`. And they worked great.
But for my usecase, I wanted to scale images while maintaining aspect ratio, so I developed my own functions to do the same.

I felt that these basic functions may be useful to many people, so I am putting my functions here, for the devs to decide wether these functions are useful enough to be included with pygame.

Starting off, a function to scale to a given width, maintaining aspect ratio
```python
def scale_w(surf, width, smooth=True):
    # Determine the height automatically and scale to aspect ratio
    height = (surf.get_height() * width) // surf.get_width()
    if smooth:
        return pygame.transform.smoothscale(surf, (width, height))
    else:
        return pygame.transform.scale(surf, (width, height))
```
The working of this basic function is pretty much self-explanatory.

Similar function, but when user wants to scale to a given height
```python
def scale_h(surf, height, smooth=True):
    # Determine the width automatically and scale to aspect ratio
    width = (surf.get_width() * height) // surf.get_height()
    if smooth:
        return pygame.transform.smoothscale(surf, (width, height))
    else:
        return pygame.transform.scale(surf, (width, height))
```

What the next function does is it scales to maintain aspect ratio, such that the scaled image fits within the given width and height. The returned surface does not guarantee that its size will be `(width, height)`, but instead, any one of either width or height is going to match the user-passed argument and the other is going to be less that the user-passed argument.

```python
def scale_wh(surf, width, height, smooth=True):
    # First, compare w1/h1 and w2/h2 (width to height ratio)
    # if w1/h1 > w2/h2 then it means that the original image was relatively
    # wider and the final image will be horizontally stretched. To maintain
    # aspect ratio, use the scale_w() function that is going to take the
    # width and adjust the final height accordingly
    if surf.get_width() * height >= surf.get_height() * width:
        return scale_w(surf, width, smooth)
    else:
        return scale_h(surf, height, smooth)
```

To better understand with an example, say I have a 80x60 image. If I want it to fit a 40x40 surface while not stretching the image, I will call
`scale_wh(image, 40, 40)` and it is going to return a 40x30 scaled image.

Now coming to the last function, I feel this function would be the most useful of all functions.
This function is similar to the above function, but it is going to scale an image (while maintaing aspect ratio, obviously) to exactly fit a given size. To do this, it is going to put padding, to center the image.
```python
def scale_aspect(surf, size, destsurf=None, padding_color=None, smooth=True):
    # Small optimisation, if initial and final sizes are aldready in ratio,
    # just scale directly
    if surf.get_width() * size[1] == surf.get_height() * size[0]:
        if destsurf is None:
            return pygame.transform.scale(surf, size)
        else:
            return pygame.transform.scale(surf, size, destsurf)
    
    # Create new surf if user has not passed destsurf argument
    if destsurf is None:
        destsurf = pygame.Surface(size)
    else:
        if destsurf.get_size() != size:
            raise ValueError("'destsurf' argument has incorrect size")

    # If user passed a padding_color, fill the surface with that.
    # Otherwise the surface is going to retain whatever was aldready there
    # in the padding region.
    if padding_color is not None:
        destsurf.fill(padding_color)
        
    # Create the temporary surface that is scaled to aspect ratio
    tempsurf = scale_wh(surf, *size, smooth)
    
    # Mark the centered position to blit the scaled surface into main surface
    blitpos = [(size[i] - tempsurf.get_size()[i]) // 2 for i in range(2)]
    
    # Blit to surf and return the final surf
    destsurf.blit(tempsurf, blitpos)
    return destsurf
```
Taking the same 80x60 image example as above, if I pass it to this function like
`scale_aspect(image, (40, 40))`, it is going to return a 40x40 surface as user requested, but the image will be 40x30, and there will be a padding of 5 pixels on top and bottom of the image.

This function also takes 2 more optional arguments if you noticed. An optional `destsurf` to make things a tiny bit faster, a new surface will be created if it is not passed and an optional `padding_color` argument. Padding color will be the color of the image padding, if it is `None` then nothing will be blitted over the padded region (if `destsurf` was passed, it is going to retain whatever it had in the padded region).

The above functions can be easily extended to smoothscale or scale function just by passing the keyword smooth argument.

Looking forward for any community feedback, any changes or improvements that need to be done can be told to me.

If my little functions look good to the devs, I would be happy to create a PR.
Note that I am not fluent with C, I will be contributing and learning at the same time, so I may need a little help to do this.

**Comment by illume** [comment]
Interesting ideas.

For API changes, it's good to get some feedback from more people.
- are the changes useful for other people? (either from pointing out existing code, or asking others)
- can existing functions be modified to support this functionality? Sometimes it's better if an existing function can do the task so we don't have too many functions over time. Maybe a new function is necessary.

Maybe ask on the mailing list, or discord #contributing channel to get some more discussion :)

**Comment by ankith26** [comment]
Of course. My idea was to get a discussion started right here on github issues. I will surely ask elsewhere to get peoples opinion.

**Comment by illume** [comment]
- We have a few different scale algorithms.
- Is alignment being centered, left/right, up/down useful?

**Comment by ankith26** [comment]
I'm soon going to create a github repo to play around with a few ideas related to this, but more importantly I am going to include examples and actual use-cases for these functions, to see wether my ideas were really good, or I got yet another useless idea that I keep getting quite often :)

From there, I will see the reaction of the community (Maybe discord, mailing list and reddit) and then get back to this issue.

**Comment by ankith26** [comment]
I have asked on reddit, for community feedback

**Comment by ankith26** [comment]
Didn’t get much of community feedback, but basically the functions I am proposing are useful, but the thing is, these functions are quite easy to implement from users end. And users can implement these functions to match their requirement.
So I’m dropping this idea.
